kong/conf_loader.lua

```diff
@@ -274,11 +438,14 @@ local CONF_INFERENCES = {
               },
   plugins = { typ = "array" },
   anonymous_reports = { typ = "boolean" },
-  nginx_daemon = { typ = "ngx_boolean" },
-  nginx_optimizations = { typ = "boolean" },
+  nginx_optimizations = {
+    typ = "boolean",
+    deprecated = {}
```

Comment:
Minor styling suggestions, `deprecated = { replacement = false }` might make this easier to understand without looking at `deprecated_properties` function.